### PR TITLE
[Shell] fixes navigate to a relative URI

### LINKED ISSUE #4971: [Shell] Cannot navigate to a relative URI
### Description

I should be able to navigate to a page without knowing the entire URI by using a relative URI. When doing this it does nothing:

```
Shell.CurrentShell.GoToAsync("/photo?payload=4.0", true);
```

or

```
Shell.CurrentShell.GoToAsync("photo?payload=4.0", true);
```

This crashes:

```
Shell.CurrentShell.GoToAsync(new ShellNavigationState(new Uri("photo?payload=4.0", UriKind.Absolute)));
```

"photo" is a route name in my Shell.xaml.

### Basic Information

- Version with issue: 4.0-pre2

### Reproduction Link

https://github.com/davidortinau/Gastropods

### PR DESCRIPTION
**Pull request by paymicro**
### Description of Change ###

Navigate to a relative URI

### Issues Resolved ### 

- fixes #4971

### API Changes ###

 
 None

### Platforms Affected ### 

- Core

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Core/Shell/Shell.cs [hunk 1]
```diff
@@ -324,6 +324,25 @@ void IShellController.UpdateCurrentState(ShellNavigationSource source)
...
+			var list = new List<string> { relativeUri.OriginalString.Trim('/') };
+			while (item != null)
+			{
+				var route = Routing.GetRoute(item)?.Trim('/');
+				if (string.IsNullOrEmpty(route))
+					break;
+				list.Insert(0, route);
+				item = item.Parent;
+			}
+			var builder = new UriBuilder(RouteScheme) { Path = string.Join("/", list) };
```
Comment (campersau) [inline Xamarin.Forms.Core/Shell/Shell.cs]: What about `Query` and `Fragment`? I think they should be preserved.

Comment (StephaneDelcroix) [reply Xamarin.Forms.Core/Shell/Shell.cs]: I agree with this

File: Xamarin.Forms.Core.UnitTests/ShellTests.cs [hunk 2]
```diff
@@ -127,7 +127,60 @@ public void SimpleGoTo()
 			Assert.That(shell.CurrentState.Location.ToString(), Is.EqualTo("app:///s/two/tabfour/content/"));
 		}
 
+		[Test]
+		public async void RelativeGoTo()
```
Comment (campersau) [inline Xamarin.Forms.Core.UnitTests/ShellTests.cs]: Should be `async Task` I guess.

**Comment by samhouts** [comment]
@paymicro Can you rebase to 3.5.0, please? Thanks!

**Comment by paymicro** [comment]
@samhouts done

**Comment by StephaneDelcroix** [comment]
build

**Pull request merged**